models/intermediate/int_intercom__latest_contact.sql

```diff
@@ -0,0 +1,34 @@
+with contact_history as (
+  select *
+  from {{ ref('stg_intercom__contact_history') }}
+),
+
+--Create an index to determine the most recent contact.
+contact_index as (
+  select
+    *,
+    row_number() over (partition by contact_id order by updated_at desc) as latest_contact_index
+  from contact_history
+),
+
+--Filter out previous contact records.
+lastest_contact as (
+  select
+    *
+  from contact_index
+  where latest_contact_index = 1
+),
+
+--Inner join on latest_contact in order to get most recent contact record without bringing through latest_contact_index.
+final as (
+  select 
+    contact_history.*
+  from contact_history
+  
+  inner join lastest_contact
```

Comment:
I'm in favor on removing this additional complexity and including the `latest_contact_index` column in the final model.  

The additional column isn't ideal, however, I think we can define it and that it won't impact the end-user experience very much.  I don't think it's worth the additional join logic (however this inner join is quite clever!)

Reply:
Sounds good, I just updated the intermediate model to be more in line with the other "latest" models. I also left a comment calling out that the index field will be passed through to the final model with minimal impact on user experience.

Let me know if you would like me to update any of it!